encoding_util.go

```diff
@@ -58,7 +58,7 @@ func (c *VncCanvas) Reset(rect *Rectangle) {
 }
 
 func (c *VncCanvas) RemoveCursor() image.Image {
-	if c.Cursor == nil || c.CursorLocation == nil {
+	if c == nil || c.Cursor == nil || c.CursorLocation == nil {
```

Comment:
Not having a canvas in your connection means that the Connect function in client.go did not succeed - you should check why...

Also, I don't see why this fix helps - since the next function to be called in client.go is canvas.PaintCursor:

canvas := c.(*ClientConn).Canvas
canvas.RemoveCursor()
parsedMsg, err := msg.Read(c)
canvas.PaintCursor()

Reply:
`c` could be `nil` if `.SetTargetImage(screenImage)` (in `example/client/main.go`) has not been called yet. because launching `ffmpeg` takes time and the message handlers are started before that in `vnc.Connect()`, `DefaultClientServerInitHandler` could be called before the renderers were given the pointer to the canvas.

I think ignoring missing canvas here could give the system just enough time to run `.SetTargetImage(screenImage)` so that by the time `PaintCursor()` is called, the renderers are ready.

See #14 for my proposed fix.